File: C/08_Math1/BOJ_09_1011_FlymetotheAlphaCentauri.c
#include <stdio.h>
#include <math.h>
main() {
	int t, cnt = 0;
	scanf("%d", &t);
	int* result = (int*)malloc(sizeof(int) * t);
	while (t--) {
		int x, y;
		scanf("%d %d", &x, &y);
		int d = y - x, max = sqrt(d);
		if (max == sqrt(d)) {
			result[cnt++] = 2 * max - 1;
		}
		else if (d <= pow(max, 2) + max) {
			result[cnt++] = 2 * max;
		}
		else {
			result[cnt++] = 2 * max + 1;
		}
	}
	for (int i = 0; i < cnt; i++) {
		printf("%d\n", result[i]);
	}
}

File: JAVA/08_Math1/BOJ_09_1011_FlymetotheAlphaCentauri.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        StringBuilder sb = new StringBuilder();
        while (t-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine(), " ");
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());
            int d = y - x, max = (int) Math.sqrt(d);
            if (max == Math.sqrt(d)) {
                sb.append(2 * max - 1).append("\n");
            } else if (d <= Math.pow(max, 2) + max) {
                sb.append(2 * max).append("\n");
            } else {
                sb.append(2 * max + 1).append("\n");
            }
        }
        System.out.print(sb);
    }
}